src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -248,9 +254,52 @@ export function QueryPageContainerComponent(): JSX.Element {
             setSubFilters(adaptedSubFilters);
             setUniqueDataValues(uniqueValuesObject);
             setTotalNrOfIsol(totalNrOfIsolates);
+            setDataIsMounted(true);
         }
     };
 
+    const fetchNewFilterValues = async (): Promise<void> => {
+        setConditionalValuesAreLoading(true);
+        const newUniqueValues = _.cloneDeep(uniqueDataValues);
+        const filterStatusList: number[] = [];
+        const getConditionalFilterValues = filter.displayedFilters.map(
+            async (mainFilter) => {
+                const selectedFilter = filter.selectedFilter[mainFilter];
+                if (_.isEmpty(selectedFilter)) {
+                    const conditionalFilterUrl = `${FILTER_URL}/${mainFilter}/${history.location.search}`;
+
+                    const filterResponse: ApiResponse<FilterConfigDTO> = await callApiService(
+                        conditionalFilterUrl
+                    );
+
+                    filterStatusList.push(filterResponse.status);
+
+                    if (filterResponse.data) {
+                        const filterProp: FilterConfigDTO = filterResponse.data;
+
+                        const uniqueValuesObject: FilterInterface = generateUniqueValuesService(
+                            filterProp
+                        );
+                        newUniqueValues[mainFilter] =
+                            uniqueValuesObject[mainFilter];
+                    }
+                }
+            }
+        );
+
+        await Promise.all(getConditionalFilterValues);
+        setUniqueDataValues(newUniqueValues);

```

Comment:
For convenience and to prevent later bugs i would not rely on the error page displayed if an api call fails. Try to keep the data always in some kind of consistent way.
Solution 1: Keep the old filters and set the filters only if status = 200 (this would still be kind of inconsistent)
Solution 2: Assign and empty object if an error occurs so that no data is available
Solution 3 (possible the best one): Keep the data with all filter values from loading the page in a separate state variable and assign it to the actual values if a call fails.

Reply:
Done. (Solution 3)